testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -147,3 +151,29 @@ def test_density_from_Universe_update_selection(self):
             self.selections['dynamic'],
             self.references['dynamic']['meandensity'],
             update_selections=True)
+
+class TestGridImport(TestCase):
+
+    def setUp(self):
+        sys.modules.pop('MDAnalysis.analysis.density', None)
+
+    @block_import('gridData')
+    def test_absence_griddata(self):
+        # if gridData package is missing an ImportError should be raised
+        # at the module level of MDAnalysis.analysis.density
+        with assert_raises(ImportError):
+            import MDAnalysis.analysis.density
+
+    def test_presence_griddata(self):
+        # no ImportError exception is raised when gridData is properly
+        # imported by MDAnalysis.analysis.density
+
+        # mock gridData in case there are testing scenarios where
+        # it is not available
```

Comment:
After a quick check, it seems that the density analysis is the only piece of the code to use GridDataFormats. The module is a dependency of MDAnalysis since version 0.7.5 where a bunch of analysis dependencies have been included as global dependencies (which I think is a rather good idea as it avoids having to install a new library for each new analysis).

I think @tylerjereddy added that test because the ``ImportError`` raised in ``analysis.density`` is customized to be more informative. That branch of the code is never called since GridDataFormats is supposedly always there. The only case I can see for GridDataFormats to be missing would be if it got uninstalled manually.

So either we let it as @tylerjereddy did it (maybe with a slightly more detailed comment), or we remove the custom ``ImportError`` (<https://github.com/MDAnalysis/mdanalysis/blob/bff425cbc672245dcfbd6f236732e213bc1e19b6/package/MDAnalysis/analysis/density.py#L102>) that should never be called.

I do not have any preference at this point.